tests/math/sequence/test_sequential_funcs.py

```diff
@@ -183,6 +183,18 @@ def test_average_weights_zero_sum():
         average(np.array([1, 2]), weights=np.array([-1, 1]))
 
 
+def test_average_scl():
+    x = mg.Tensor(np.array([1, 2, 3, 4]))
+    weights = mg.Tensor(np.array([1, 2, 3, 4]))
+    avg, scl = average(x, weights=weights, returned=True)
+    expected_avg, expected_scl = np.average(x.data, weights=weights.data, returned=True)
+    assert np.isclose(avg.data, expected_avg)
+    assert np.isclose(scl.data, expected_scl)
+    scl.backward()
+    assert avg.grad is None
+    assert weights.grad is not None
```

Comment:
You should add a check that the gradient for the weights is indeed correct.

Reply:
Good call, like you pointed out a branch of `average` was completely untested where `scl` was being read even though it was not defined.